segmentation/utils/spatial_analysis_utils.py

```diff
@@ -33,9 +33,11 @@ def calc_dist_matrix(label_map):
     return dist_mats_xr
 
 
-def compute_close_cell_num(patient_data_markers, label_idx, thresh_vec,
-                           dist_mat, marker_num, dist_lim):
+def compute_close_cell_num(dist_mat, dist_lim, num, analysis_type, cell_label_idx=None,
```

Comment:
This function has too many arguments, which makes it hard to understand. Let's change the logic as follows.

Rather than subsetting the data beforehand, we'll do the subsetting inside this function. So instead of patient_data, pheno_codes, patient_data_markers, etc, all getting passed in, we'll just pass in a single array that has all of the data for that patient. Then, in the new function underneath, we'll subset appropriately.